tests/functional/ipset_controller_test.go

```diff
@@ -312,6 +314,72 @@ var _ = Describe("IPSet controller", func() {
 		})
 	})
 
+	When("an IPSet with multiple networks gets created", func() {
+		var netSpecs []networkv1.Network
+		var networks []networkv1.IPSetNetwork
+		BeforeEach(func() {
+			netSpecs = []networkv1.Network{}
+			netSpecs = append(netSpecs, GetNetSpec(net1, GetSubnet1(subnet1)))
+			netSpecs = append(netSpecs, GetNetSpec(net2, GetSubnet2(subnet1)))
+			netSpecs = append(netSpecs, GetNetSpec(net3, GetSubnet3(subnet1)))
+			networks = []networkv1.IPSetNetwork{}
+			networks = append(networks, GetIPSetNet1Lower())
+			networks = append(networks, GetIPSetNet2())
+			networks = append(networks, GetIPSetNet3())
+
+			Expect(networks).To(HaveLen(len(netSpecs)))
```

Comment:
don't think this is a must. like you could have more networks in the netSpecs (osnetconfig) then in the ipset networks you request IPs for.